sssd_test_framework/hosts/keycloak.py

```diff
@@ -0,0 +1,111 @@
+"""Keycloak multihost host."""
+
+from __future__ import annotations
+
+import time
+
+from pytest_mh.ssh import SSHProcessError
+
+from .base import BaseDomainHost
+
+__all__ = [
+    "KeycloakHost",
+]
+
+
+class KeycloakHost(BaseDomainHost):
+    """
+    Keycloak host object.
+
+    Provides features specific for Keycloak server.
+
+    This class adds ``config.adminpw`` multihost configuration option to set
+    password of the Keycloak admin user so we run kcadm.sh commands to set
+    options in Keycloak.
+
+    .. code-block:: yaml
+        :caption: Example multihost configuration
+        :emphasize-lines: 4
+
+        - hostname: master.keycloak.test
+          role: keycloak
+          config:
+            adminpw: Secret123
+
+    .. note::
+
+        Full backup and restore is supported. However, the operation relies on
+        ``kc.sh export`` and ``kc.sh import`` commands which can take several
+        seconds to finish.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.adminpw = self.config.get("adminpw", "Secret123")
+
+        # Backup of original data
+        self.__backup: str | None = None
+
+    def kclogin(self) -> None:
```

Comment:
I modeled this after the IPA hosts class which has a kinit for admin.  This is basically the same.   

That being said, your suggestion would make the login more generally useful as I believe this could be used for login tests against keycloak for any user.   Should that be here or in a utils class like authentication?  Or could that wait for a later update when we add Keycloak specific tests as well?

Reply:
Does the function need to happen on the client or keycloak server? keycloak, here. Client, utils, under authentication probably like kerberos.

Reply:
Keycloak server.

Reply:
They should be in the keycloak role, then. The host, only really contains backup, restore methods and passes variables to the role.

Reply:
I don't know, all the servers we use are containers, so the setup is already done, we're just passing variables to configure them from hosts. 

I think install/setup should be in hosts, under init. When the host is selected the service is already configured. So maybe we remove kclogin and make dump it in init instead?

Reply:
It's used in the keycloak role setup() the same way the ipa role uses the ipa hosts kinit in the ipa role setup.  I just followed that pattern.   I do believe it's needed every time the role is instantiated to make sure the login is done before running the kcadm commands against keycloak.

Reply:
Scott chatted about this I misunderstood the purpose of this function.